test/transaction/transaction-sidechain.js

```diff
@@ -1,122 +1,121 @@
+'use strict';
+
+const assert = require('chai').assert;
+const Transaction = require('../../lib/transaction');
+const txes = require('../data_tx/tx.json');
+
+const revert = function (pubkey) {
+  let pubkeyhash = '';
+  let i = pubkey.length - 1;
+  while (i > 0) {
+    pubkeyhash += pubkey[i - 1] + pubkey[i];
+    i = i - 2;
+  }
+  return pubkeyhash;
+};
+
+describe('#Sidechain transactions', function () {
+
+  describe('SC_create/FT/VMBTR', function () {
+    let index = 0;
+    txes.forEach(function (ele) {
+
+      it('vector #' + index + ': ' + ele.description, function () {
+
+        const hexa = ele.hex;
+        const tx = Transaction(hexa);
+        const txJson = ele.json;
+        //version
+        assert.equal(tx.version, txJson.version);
+
+        //txid
+        assert.equal(tx.hash, txJson.txid);
+
+        //locktime
+        assert.equal(tx.nLockTime, txJson.locktime);
+
+        //vin
+        assert.equal(tx.inputs.length, txJson.vin.length);
+        for (let i = 0; i < txJson.vin.length; i++) {
+          assert.equal(tx.inputs[i].toObject().prevTxId, txJson.vin[i].txid);
+          assert.equal(tx.inputs[i].toObject().sequenceNumber, txJson.vin[i].sequence);
+          assert.equal(tx.inputs[i].toObject().outputIndex, txJson.vin[i].vout);
+        }
+
+        //vout
+        assert.equal(tx.outputs.length, txJson.vout.length);
+        for (let i = 0; i < txJson.vout.length; i++) {
+          assert.equal(tx.outputs[i].toObject().satoshis, txJson.vout[i].valueZat);
+        }
+
+        //Sidechain
+        const scParamsObject = tx.sc_params.toObject();
+        const sc = {
+          vsc_ccout: scParamsObject.vsc_ccout,
+          vft_ccout: scParamsObject.vft_ccout,
+          vcsw_ccin: scParamsObject.vcsw_ccin,
+          vmbtr_out: scParamsObject.vmbtr_out
+        };
+
+        //vcsw
+        assert.equal(sc.vcsw_ccin.length, txJson.vcsw_ccin.length);
+        for (let i = 0; i < txJson.vcsw_ccin.length; i++) {
+          assert.equal(sc.vcsw_ccin[i].value / 1e8, txJson.vcsw_ccin[i].value);
+          assert.equal(sc.vcsw_ccin[i].scId, txJson.vcsw_ccin[i].scId);
+          assert.equal(sc.vcsw_ccin[i].nullifier, txJson.vcsw_ccin[i].nullifier);
+          assert.equal(sc.vcsw_ccin[i].scProof, txJson.vcsw_ccin[i].scProof);
+          assert.equal(sc.vcsw_ccin[i].actCertDataHash, txJson.vcsw_ccin[i].actCertDataHash);
+          assert.equal(sc.vcsw_ccin[i].ceasingCumScTxCommTree, txJson.vcsw_ccin[i].ceasingCumScTxCommTree);
+          assert.equal(sc.vcsw_ccin[i].redeemScript, txJson.vcsw_ccin[i].redeemScript.hex);
+          assert.equal(sc.vcsw_ccin[i].pubKeyHash, revert(txJson.vcsw_ccin[i].scriptPubKey.asm.split(' ')[2]));
+        }
+
+        //vsc_ccout
+        assert.equal(sc.vsc_ccout.length, txJson.vsc_ccout.length);
+        for (let i = 0; i < txJson.vsc_ccout.length; i++) {
+          assert.equal(sc.vsc_ccout[i].epoch_length, txJson.vsc_ccout[i]['withdrawal epoch length']);
```

Comment:
This field will be renamed in zendoo to "withdrawalEpochLength".

Reply:
It doesn't affect the code, just this test (no changes needed)